tests/plugin_tests/test_weather.py

```diff
@@ -0,0 +1,46 @@
+import math
+
+import pytest
+
+
+@pytest.mark.parametrize('bearing,direction', {
+    (360, 'N'),
+    (0, 'N'),
+    (1, 'N'),
+    (15, 'NNE'),
+    (30, 'NNE'),
+    (45, 'NE'),
+    (60, 'ENE'),
+    (75, 'ENE'),
+    (90, 'E'),
+    (105, 'ESE'),
+    (120, 'ESE'),
+    (135, 'SE'),
+    (150, 'SSE'),
+    (165, 'SSE'),
+    (180, 'S'),
+    (348.75, 'N'),
+    (348.74, 'NNW'),
+})
+def test_wind_direction(bearing, direction):
+    from plugins.weather import bearing_to_card
+    assert bearing_to_card(bearing) == direction
+
+
+@pytest.mark.parametrize('temp_f,temp_c', [
+    (32, 0),
+    (212, 100),
+    (-40, -40),
+])
+def test_temp_convert(temp_f, temp_c):
+    from plugins.weather import convert_f2c
+    assert convert_f2c(temp_f) == temp_c
```

Comment:
This MAY break due to floating point errors on some systems. Not sure if it should be changed due to that though

Reply:
Nowhere do I touch floating point, I use Fraction() for a reason.